services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -182,7 +173,11 @@ public VeniceServer(VeniceServerContext ctx) throws VeniceException {
     this.metricsRepository = ctx.getMetricsRepository();
     this.icProvider = ctx.getIcProvider();
     this.serviceDiscoveryAnnouncers = ctx.getServiceDiscoveryAnnouncers();
-    this.pubSubClientsFactory = Objects.requireNonNull(ctx.getPubSubClientsFactory(), "PubSubClientsFactory is null");
+    VeniceServerConfig veniceServerConfig = ctx.getVeniceConfigLoader().getVeniceServerConfig();
+    this.pubSubClientsFactory = new PubSubClientsFactory(
+        veniceServerConfig.getPubSubProducerAdapterFactory(),
+        veniceServerConfig.getPubSubConsumerAdapterFactory(),
+        veniceServerConfig.getPubSubAdminAdapterFactory());
     this.sslFactory = Optional.ofNullable(ctx.getSslFactory());
```

Comment:
let's get `PubSubClientsFactory` from server config instead of individual factories